src/cpp/src/visual_language/inputs_embedder.cpp

```diff
@@ -112,19 +128,30 @@ class InputsEmbedder::IInputsEmbedder {
                 new_templated_chat_history = m_tokenizer.apply_chat_template(m_history, add_generation_prompt, chat_template_fallback);
             }
             ov::Tensor new_chat_tokens = m_tokenizer.encode(new_templated_chat_history).input_ids;
-            if (m_is_cache_empty) {
+            TokenizedInputs prev_chat_tokens = m_tokenizer.encode(m_templated_chat_history);
+
+            // some symbols combinations can be encoded by the tokenizer in different ways
+            // if we met sequence with such combination of symbols, we cannot correctly subtract the new history from the old history
+            // and find the difference as a prompt
+            // so let's check it out and use the whole history in this case
+            if (!m_is_cache_empty && m_trust_encoded_history) {
```

Comment:
!m_is_cache_empty  => !m_tokenized_chat_history.empty()

looks like we can drop `m_is_cache_empty` flag similarly to LLMPipeline